Zend/tests/magic_methods_011.phpt

```diff
@@ -0,0 +1,11 @@
+--TEST--
+__set first argument should be a string when typed
+--FILE--
+<?php
+
+class Foo {
+	function __set(\Countable $name, $value) {}
+}
+?>
+--EXPECTF--
+Fatal error: Foo::__set(): Argument #0 ($name) must be of type string when declared in %s on line %d
```

Comment:
`Argument #0` is an off-by-one error :)

Reply:
Technically, `argument` is not right here, since these are parameters. I'm not sure what the best message format would be, though.

Reply:
This is the standard today: https://3v4l.org/ouEc3.

Let's revisit it for PHP 8.1.

Reply:
But these are completely different cases: Argument passed to functions (as in the 3v4l example), while are parameter are declared. Since these are errors related to the function signature, arguments just doesn't make sense in this context.

Reply:
Going through zend_compile, we do seem to be fairly consistent in using "parameter" in this context. So I agree that this should be 
```suggestion
Fatal error: Foo::__set(): Parameter #1 ($name) must be of type string when declared in %s on line %d
```
The "must be of type string" part is not quite correct, because it can also be a supertype, but I think it's better to keep the message as-is than trying to be fully precise here.